core/src/main/java/novoda/lib/sqliteprovider/analyzer/IndexColumnsQuery.java

```diff
@@ -0,0 +1,26 @@
+package novoda.lib.sqliteprovider.analyzer;
+
+import android.database.Cursor;
+
+class IndexColumnsQuery implements Query<String> {
+
+    private final String indexName;
+
+    public IndexColumnsQuery(String indexName) {
+        this.indexName = indexName;
+    }
+
+    @Override
+    public String getSqlStatement() {
+        return "PRAGMA index_info('" + indexName + "');";
+    }
+
+    @Override
+    public String parseRow(Cursor cursor) {
+        return getColumnName(cursor);
+    }
+
+    private String getColumnName(Cursor cursor) {
```

Comment:
I'd probably inline this method and instead have a constant like `NAME_COLUMN_INDEX`.